packages/app/src/renderer/wrappers/base-app.component.tsx

```diff
@@ -101,41 +63,12 @@ const BaseApp: FunctionComponent<Props> = ({
     }
   }, [deviceParred, settingsLoaded])
 
-  const allowToAppCollectingData = (): void => {
-    toggleAppCollectingData(true)
-  }
-
-  const disallowToAppCollectingData = (): void => {
-    toggleAppCollectingData(false)
-  }
-
-  const closeAppUpdateStepModal = (): void => {
-    setAppUpdateStepModalDisplayed()
-    toggleAppUpdateStepModalShow(false)
-  }
-
   return (
     <>
       <NetworkStatusChecker />
-      <CollectingDataModal
-        open={collectingDataModalVisible}
-        onActionButtonClick={allowToAppCollectingData}
-        closeModal={disallowToAppCollectingData}
-      />
-      {appUpdateRequired && (
-        <AppUpdateStepModal
-          forced
-          appLatestVersion={appLatestVersion}
-          appCurrentVersion={appCurrentVersion}
-        />
-      )}
-      {!appUpdateRequired && appUpdateStepModalVisible && (
-        <AppUpdateStepModal
-          appLatestVersion={appLatestVersion}
-          appCurrentVersion={appCurrentVersion}
-          closeModal={closeAppUpdateStepModal}
-        />
-      )}
+      <CollectingDataModal />
```

Comment:
Last, but not last. What do you think about implementing a single container for all modals?

Reply:
I think about it, it could be a wrap component as part of `modalsManager`, what do you think?

Reply:
https://github.com/mudita/mudita-center/pull/558/commits/4bd3b5f218f7f2d92b912f294ae4e4e61ca0a29d